autoscaler/controllers/actions/k8sattributesresolver_controller.go

```diff
@@ -217,29 +279,38 @@ func k8sAttributeConfig(ctx context.Context, k8sclient client.Client, namespace
 		collectContainer = collectContainer || config.CollectContainerAttributes
 		collectClusterUID = collectClusterUID || config.CollectClusterUID
 
-		// Add label attributes, newer configs override older ones with same Tag
+		// Add label attributes
+		// For each label, we may have multiple sources (e.g., pod and namespace).
+		// We need to generate separate processor config entries for each source,
+		// with the correct precedence order (lower precedence sources processed first,
+		// so higher precedence sources can override).
 		for _, label := range config.LabelsAttributes {
-			from := actionv1.PodAttributeSource
-			if label.From != nil {
-				from = actionv1.K8sAttributeSource(*label.From)
-			}
-			labelAttributes[label.LabelKey] = k8sTagAttribute{
-				Tag:  label.AttributeKey,
-				Key:  label.LabelKey,
-				From: string(from),
+			sources := getEffectiveSources(label.From, label.FromSources)
+			for _, source := range sources {
+				// Use a composite key of LabelKey and source to allow multiple sources for the same label
+				key := label.LabelKey + ":" + string(source)
+				labelAttributes[key] = k8sTagAttribute{
```

Comment:
wondering why `labelAttributes` is a map and not list. is it for removing duplicate conflicting configurations if they are set by the user?

Reply:
Theoretically it shouldn't be possible for the user, but because we're trusting yamls this makes sure you only get one value per label per source